pkg/image/api/helper.go

```diff
@@ -184,7 +184,38 @@ func ParseDockerImageReference(spec string) (DockerImageReference, error) {
 		ref.ID = id
 		break
 	default:
-		// TODO: this is no longer true with V2
+		// Handle multiple segments in form: registry/namespace/namesegment1/namesegment2/.../name
+		// but not this form: http://registry/namespace/name. We don't support using the schema in
+		// the DockerImageReference.
+		if len(repoParts) > 3 && !strings.HasSuffix(repoParts[0], ":") {
+			if IsRegistryDockerHub(repoParts[0]) {
+				// DockerHub doesn't support multiple segments.
+				return ref, fmt.Errorf("the docker pull spec %q must be of the form / to be valid for the docker hub", spec)
```

Comment:
@ncdc I added this condition because we have previously banned more than 3 parts, and by this we banned `docker.io/foo/bar/baz` as well. I wanted to allow multiple segments only for custom `docker/distribution` servers.